src/network/networkd-manager.c

```diff
@@ -621,6 +623,13 @@ Manager* manager_free(Manager *m) {
         HASHMAP_FOREACH(link, m->links_by_index)
                 (void) link_stop_engines(link, true);
 
+        HASHMAP_FOREACH_KEY(v, k, m->sysctls) {
+                (void) free(k);
+                (void) free(v);
+        }
```

Comment:
please allocate the hashmap with a hash_ops of `string_hash_ops_free_free`. Then you can drop this loop as hashmap_free() will auto-destroy the contents for you

Reply:
I just noticed that `path_hash_ops_free_free` exists, isn't it even better?